.github/workflows/static_analysis.yaml

```diff
@@ -39,33 +35,50 @@ jobs:
         #  --max-configs=6  :   Using less configuration permutations (default is 12) to reduce run time. Detects less errors. Removed.
         #  -Dxxx            :   preprocessor configuration to use. Relevant flags taken from build on Ubuntu.
         run: > 
-          cppcheck ./src ./include ./common ./tools ./examples ./third-party/realdds ./third-party/utilities
+          cppcheck ./src ./include ./common ./tools ./examples ./third-party/realdds ./third-party/rsutils
           --quiet --std=c++11 --xml -j4 --enable=warning
-          -I./src -I./include -I./third-party/utilities/include -I./common
+          -I./src -I./include -I./third-party/rsutils/include -I./common
           -i./src/mf -i./src/uvc -i./src/win -i./src/winusb --suppress=*:third-party/json.hpp
           -DBUILD_WITH_DDS -DHWM_OVER_XU -DRS2_USE_V4L2_BACKEND -DUNICODE -DUSING_UDEV -DCHECK_FOR_UPDATES -D__linux__
           &> cppcheck_run.log
 
-      - name: Upload log
+      - name: Diff
+        id: diff-step
+        continue-on-error: true 
+        shell: bash
+        run: |
+          python3 .github/workflows/cppcheck-parse.py --severity E cppcheck_run.log | sort --key 3 > cppcheck_run.parsed.log
+          python3 .github/workflows/cppcheck-parse.py --severity E .github/workflows/cppcheck_run.log | sort --key 3 > cppcheck_run.parsed.golden
+          diff cppcheck_run.parsed.golden cppcheck_run.parsed.log
+
+      - name: Upload logs
         uses: actions/upload-artifact@v3
         with: 
           name: cppcheck_log
-          path: cppcheck_run.log
-
+          path: |
+            cppcheck_run.log
+            cppcheck_run.parsed.log
```

Comment:
Is the parsed log supposed to be uploaded? I did not see it saved in your last run GH actions.
I don't think that we do need to save it, we can always run the script to parse `cppcheck_run.log` if needed.

Reply:
Right. But it's uploaded because it doesn't cost us anything and it makes it easier to look at without knowing how to run the script.